delfin/drivers/dell_emc/vnx_block/component_handler.py

```diff
@@ -0,0 +1,490 @@
+# Copyright 2020 The SODA Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import six
+from oslo_log import log
+from oslo_utils import units
+
+from delfin import exception
+from delfin.common import constants
+
+LOG = log.getLogger(__name__)
+
+
+class ComponentHandler():
+
+    EMCVNX_VENDOR = 'DELL EMC'
+    RAID_GROUP_ID_PREFIX = 'raid_group_'
+
+    STATUS_MAP = {'Ready': constants.StoragePoolStatus.NORMAL,
+                  'Offline': constants.StoragePoolStatus.OFFLINE,
+                  'Valid_luns': constants.StoragePoolStatus.NORMAL,
+                  'Bound': constants.StoragePoolStatus.NORMAL}
+
+    VOL_TYPE_MAP = {'no': constants.VolumeType.THICK,
```

Comment:
Are these VNX specific?

Reply:
Yes, use the command execution in the vnx block command-line document to obtain the status values in Lun and raidgroup.

Reply:
ok